modules/javafx.base/src/main/java/javafx/beans/value/ObservableValue.java

```diff
@@ -251,4 +252,57 @@ default ObservableValue<T> orElse(T constant) {
     default <U> ObservableValue<U> flatMap(Function<? super T, ? extends ObservableValue<? extends U>> mapper) {
         return new FlatMappedBinding<>(this, mapper);
     }
+
+    /**
+     * Returns an {@code ObservableValue} that holds this value and is updated only
+     * when {@code condition} holds {@code true}.
+     * <p>
+     * The returned {@code ObservableValue} only observes this value when
+     * {@code condition} holds {@code true}. This allows this {@code ObservableValue}
+     * and the conditional {@code ObservableValue} to be garbage collected if neither is
+     * otherwise strongly referenced when {@code condition} holds {@code false}.
+     * <p>
+     * A currently observed binding will observe its source, which means it will not be eligible
+     * for garbage collection while its source isn't. However, when using {@code when} this {@code ObservableValue}
+     * can still be eligible for garbage collection when {@code condition} holds {@code false} and {@code condition}
+     * itself is also eligible for garbage collection.
```

Comment:
I think that these 2 paragraphs can be combined. There is some repetition in the last sentence ("However...") of the 1st paragraph. Maybe add to the 1st paragraph at the end:

> This is in contrast to the general behavior of bindings, where the binding is never eligible for GC as long as the source isn't.

If you think it's important to note that. Then remove the 2nd paragraph?

Reply:
Yes, I've changed it.  I had to rephrase it to: "This is in contrast to the general behavior of bindings, where the binding is only eligible for garbage collection when not observed itself." -- with `when` there are two options for allowing GC, the one I just mentioned and when `condition` is `false` and not referenced itself.